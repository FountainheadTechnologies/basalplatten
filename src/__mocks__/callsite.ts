module.exports = () => ([{
}, {
  getFileName: () => 'test-caller.js'
}]);
// @hack: Make module look like es6 to prevent mangling
module.exports.__esModule = true;
