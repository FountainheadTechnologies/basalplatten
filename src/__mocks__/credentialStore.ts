export const credentialStore = {
  setToken: jest.fn(),
  getToken: jest.fn(),
  clearToken: jest.fn()
}
