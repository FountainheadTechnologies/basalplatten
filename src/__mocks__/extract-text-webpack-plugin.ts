module.exports = class ExtractTextPlugin {
  extract() {}
};
// @hack: Make module look like es6 to prevent mangling
module.exports.__esModule = true;
