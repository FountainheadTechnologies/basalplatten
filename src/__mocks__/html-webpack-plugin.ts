module.exports = class HtmlWebpackPlugin {};
// @hack: Make module look like es6 to prevent mangling
module.exports.__esModule = true;
