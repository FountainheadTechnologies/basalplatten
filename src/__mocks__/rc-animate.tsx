import React from 'react';

const RcAnimate = props => (
  <div>{props.component}</div>
)

RcAnimate.displayName = 'RcAnimate';

export default RcAnimate;
