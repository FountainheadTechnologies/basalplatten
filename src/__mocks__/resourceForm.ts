const submit = jest.fn();

export const createToken = {
  schema: {
    required: ['username', 'password'],
    properties: {
      username: { type: 'string' },
      password: { type: 'string' }
    },
  },

  submit
}

export const createPerson = {
  schema: {
    required: ['forename', 'surname'],
    properties: {
      forename: { type: 'string' },
      surname: { type: 'string' },
      date_of_birth: {
        type: 'string',
        format: 'date-time'
      }
    }
  },

  submit
}

export const updatePerson = {
  schema: {
    required: [],
    default: {
      forename: 'Test',
      surname: 'Person',
      date_of_birth: '1985-08-02T23:00:00.000Z'
    },
    properties: {
      forename: { type: 'string' },
      surname: { type: 'string' },
      date_of_birth: {
        type: 'string',
        format: 'date-time'
      }
    }
  }
}
