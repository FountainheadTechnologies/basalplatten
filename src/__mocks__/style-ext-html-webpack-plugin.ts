module.exports = class StyleExtHtmlWebpackPlugin { };
module.exports.inline = () => { };
// @hack: Make module look like es6 to prevent mangling
module.exports.__esModule = true;
