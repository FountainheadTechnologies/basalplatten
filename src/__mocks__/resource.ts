import { createToken } from './resourceForm';

export const api = {
  properties: {},
  meta: {},

  config: {
    headers: {}
  },

  formNamed(rel: string, name: string) {
    if (rel === 'token' && name === 'create') {
      return createToken;
    }
  }
}
