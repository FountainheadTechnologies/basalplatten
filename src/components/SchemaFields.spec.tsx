import React from 'react';
import { shallow } from 'enzyme';

import { SchemaFields } from './SchemaFields';
import { createPerson } from '../__mocks__/resourceForm';

const context = {
  schema: createPerson.schema
}

describe('default behaviour', () => {
  const schemaFields = shallow(<SchemaFields />, { context })

  it('renders SchemaField components for all properties in schema', () => {
    expect(schemaFields).toMatchSnapshot();
  });
});

describe('when `props.only` is set', () => {
  const schemaFields = shallow((
    <SchemaFields only={['forename', 'surname']} />
  ), { context })

  it('only renders SchemaField components for properties in schema and in `props.only`', () => {
    expect(schemaFields).toMatchSnapshot();
  });
});

describe('when `props.except` is set', () => {
  const schemaFields = shallow((
    <SchemaFields except={['date_of_birth']} />
  ), { context })

  it('renders SchemaField components for all properties in schema not in `props.except`', () => {
    expect(schemaFields).toMatchSnapshot();
  });
});

describe('when `props.except` and `props.only` are set', () => {
  it('throws an error', () => {
    expect(() => {
      shallow((
        <SchemaFields
          only={['forename']}
          except={['date_of_birth']}
        />
      ), { context });
    }).toThrowError('Props `only` and `except` are mutually exclusive');
  })
});
