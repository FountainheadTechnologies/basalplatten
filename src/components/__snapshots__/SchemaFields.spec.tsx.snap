// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default behaviour renders SchemaField components for all properties in schema 1`] = `
<div
  className="schema-fields"
>
  <Component
    key="forename"
    label="Forename"
    name="forename"
  />
  <Component
    key="surname"
    label="Surname"
    name="surname"
  />
  <Component
    key="date_of_birth"
    label="Date Of Birth"
    name="date_of_birth"
  />
</div>
`;

exports[`when \`props.except\` is set renders SchemaField components for all properties in schema not in \`props.except\` 1`] = `
<div
  className="schema-fields"
>
  <Component
    key="forename"
    label="Forename"
    name="forename"
  />
  <Component
    key="surname"
    label="Surname"
    name="surname"
  />
</div>
`;

exports[`when \`props.only\` is set only renders SchemaField components for properties in schema and in \`props.only\` 1`] = `
<div
  className="schema-fields"
>
  <Component
    key="forename"
    label="Forename"
    name="forename"
  />
  <Component
    key="surname"
    label="Surname"
    name="surname"
  />
</div>
`;
